### ISSUE TITLE
esm_datastore loaded from yaml catalog send Error : got an unexpected keyword argument 'metadata'

### ISSUE THREAD

**Issue by ltroussellier** [intake/intake#703]
Versions : 

intake                         0.6.6                        pypi_0    pypi
intake-esm                2022.9.18                pypi_0    pypi

I got a YAML intake catalog "MasterCat" in witch there are multiple esm-datastore sources. This was working with previous versions, but now it does not. 

part of MasterCat.yaml : 

```
description: 'master catalog for all esm catalogs available'

plugins:
  source:
    - module: intake_esm

sources:
  PMIP3:
    args:
      urlpath: "catalogs/PMIP3.json"
    description: 'PMIP3 catalog Storage'
    driver: intake_esm.esm_datastore
    metadata: {}
```

When the PMIP3 esm_datastore is loaded with : 
 ```
import intake
pmip3_cat = intake.open_esm_datastore("catalogs/PMIP3.json")
```
everything is fine 

when this catalog is loaded through the yaml catalog : 
```
import intake
catMaster = intake.open_catalog("cat/master.yml")
pmip3 = catMaster["PMIP3"]

```
```
TypeError: esm_datastore.__init__() got an unexpected keyword argument 'metadata'
```
It seems that something changed that broke my catalog, i understand that. But after some investigation, i can't get it working 
what im doing wrong ? 
what parameters i have to change in MasterCat in order to let intake correctly load my esm catalog ? 

Thanks in advance

Already Post here : https://github.com/intake/intake-esm/issues/567 but no response in 3 days so i try here

**Comment by martindurant** [comment]
Can you omit the `metadata: {}` line? All intake sources, including catalogs, should really allow for metadata, but catalogs need it less and ESM doesn't support it. If my suggestion doesn't fix it, we can fix it in intake_esm.

**Comment by ltroussellier** [comment]
Unfortunately, even if i remove the `metadata: {}`, it doesn't change the error. My guess is that there is not enough arguments in the masterCat, and the default driver send an empty  "metadata" argument through esm_datastore.init() method.

**Comment by ltroussellier** [comment]
I tried to change the __init__ method from esm_datastore class like the commit change. It doesn't work but there is some progress : TypeError: esm_datastore.__init__() missing 1 required positional argument: 'obj'
I also tried to uninstall and re install from github, same Error. 
I understand that i have to use the github version until the next release (its OK) but now, i have another issue to investigate.
